test/routes-registered.test.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const { test } = require('tap')
+const FindMyWay = require('../')
+
+function initializeRoutes (router, handler, quantity) {
+  for (const x of Array(quantity).keys()) {
+    router.on('GET', '/test-route-' + x, handler)
+  }
+  return router
+}
+
+test('verify routes registered', t => {
+  const quantity = 5
+  // 1 (check length) + quantity of routes * quantity of tests per route
+  t.plan(1 + (quantity * 1))
+
+  let findMyWay = FindMyWay()
+  const defaultHandler = (req, res, params) => res.end(JSON.stringify({ hello: 'world' }))
+
+  findMyWay = initializeRoutes(findMyWay, defaultHandler, quantity)
+  t.strictEqual(findMyWay.routes.length, quantity)
+  findMyWay.routes.map((route, idx) => {
+    t.deepEqual(route, {
+      method: 'GET',
+      path: '/test-route-' + idx,
+      opts: {},
+      handler: defaultHandler,
+      store: undefined
+    })
+  })
+})
+
+test('verify routes registered and deregister', t => {
+  // 1 (check length) + quantity of routes * quantity of tests per route
+  t.plan(2)
+
+  let findMyWay = FindMyWay()
+  const quantity = 2
+  const defaultHandler = (req, res, params) => res.end(JSON.stringify({ hello: 'world' }))
+
+  findMyWay = initializeRoutes(findMyWay, defaultHandler, quantity)
+  t.strictEqual(findMyWay.routes.length, quantity)
+  findMyWay.off('GET', '/test-route-0')
+  t.notEqual(findMyWay.routes.length, quantity)
```

Comment:
This is not right, `off` should remove the route from the `routes` array. Now is not doing it because `routes` was not a public API, but we should update that code as well.

Reply:
`findMyWay.off` remove route from `findMyWay.routes` array. I don't understand

Reply:
Maybe I'm wrong, but from the test it appears that after calling `off`, the length of the routes array is still 2.

Reply:
No, the length is decreased. I'll change the assert to better explain.

Reply:
Ooops! I misread the not equal :D
Thanks!